### ISSUE TITLE
Logstash don't exit after click ctl+c in ubuntu12.10

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-input-redis#9]
(This issue was originally filed by @leiding at https://github.com/elastic/logstash/issues/2143)

---

step 1: start logstash 
./bin/logstash agent -f index.conf
Using milestone 2 input plugin 'redis'. This plugin should be stable, but if you see strange behavior, please let us know! For more information on plugin milestones, see http://logstash.net/docs/1.4.2/plugin-milestones {:level=>:warn}

Step2: Try to exit with cto+c, but the cursor can't return to the shell.

^CInterrupt received. Shutting down the pipeline. {:level=>:warn}

cat index.conf
input {
    redis
    {
    host => "10.111.96.175"
    data_type =>"list"
    port => "6379"
    key => "logstash"
    type => "redis-input"
    }

}

output
    {
    elasticsearch {
    host => "10.111.96.175"
    port => "9300"
    }
}

**Comment by guyboertje** [comment]
I don't think this bug exists anymore after implementing the pipeline stop in LS v2.

**Comment by opnarius** [comment]
I can still replicate this with `logstash 2.2.2` and Ubuntu 14.04 versions. 

This is my config:

```
input {
    redis {
        host => "redis01"
        port => 6379
        data_type => "pattern_channel"
        key => "*.firehose"
        add_field => {
            type => "firehose"
        }
    }
}
```
The issue happens when I run logstash directly in console or as a service. With a service, I have to `kill -9` to stop logstash.

Is there anything I can try to help with troubleshooting this?

**Comment by guyboertje** [comment]
@opnarius - thanks for the update. I will look into this. I have not really investigated the pattern channel data type with respect to LS quit.
